docs/specs/mlnet-cli/MLNET-CLI-Specs.md

```diff
@@ -0,0 +1,969 @@
+
+```console
+dotnet tool install -g mlnet
+```
+
+## Using the tool
+
+Once the tool is installed, you can call it by using its command. Note that the command may not be the same as the package name.
+
+You can use it with:
+
+```console
+mlnet
+```
+
+## Command 'new'
+
+`mlnet new` will create the following assets:
+
+- A new project or group of projects (by default, a single console app) with the generated .NET code.
+- The "best model" (or multiple "best models") as serialized files.
+- An analysis report of the generated models. 
+
+By default (with no app template parameter), the CLI will create a single console application containing either the training code and the model scoring code. However, in next ML.NET CLI versions (heading v1.0), when the user provides an app-type template argument (such as `mlnet new web`), it will be able to generate different application type projects (web, console, desktop, Unity, etc.) for the model scoring code. 
+
+## Examples
+
+Simplest CLI command for a binary classification problem (AutoML will need to infer most of the configuration from the provided data):
+
+(*Release 0.1 examples*)
+
+` mlnet new --ml-task BinaryClassification --dataset "/MyDataSets/Cars.csv"`
+
+Another simple CLI command for a regression problem where the target-column is provided explecitely:
+
+` mlnet new --ml-task Regression --dataset "/MyDataSets/Sales.csv" --label-column "MonthlySales" `
+
+Create and train a binary-classification model with quite a few explicit arguments:
+
+` mlnet new --ml-task BinaryClassification --app-type console --dataset "/MyDataSets/Population-Training.csv" --test-dataset "/MyDataSets/Population-Test.csv" --label-column-name "InsuranceRisk" --max-exploration-time 600`
+
+Create and train a model based on arguments specified in the CLI response file (.rsp) with arguments containing the configuration:
+
+` mlnet new @my_cli_config_args.rsp`
+
+
+(*Release 0.2 examples*)
+
+Simplest command where the tool infers the type of ML taks to perform based on the data:
+
+` mlnet new --dataset "/MyDataSets/Cars.csv"`
+
+Create and train a model based on parameters specified in the .rsp file plus more advanced model settings in the .yaml file:
+
+` mlnet new @my_cli_config_args.rsp --model-settings-file "./my_model_settings.yaml"  `
+
+
+(*Release v1.0 examples*)
+
+CLI command specifying to run AutoML compute in Azure's cloud 
+
+` mlnet new --ml-task BinaryClassification --dataset "/MyDataSets/Cars.csv" --label-column-name "InsuranceRisk" --azure-automl-service-key "key-value"`
+
+
+
+Syntax should be aligned to [dotnet new](https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-new?tabs=netcore21#synopsis).
+
+#### Syntax
+
+```console
+mlnet new 
+
+--------------- (v0.1) -------------------
+
+--ml-task <value>
+
+--dataset <value>
+[
+ [--validation-dataset <value>]
+  --test-dataset <value>
+]
+
+--label-column-name <value>
+|
+--label-column-index <value>
```

Comment:
must add feature, groupid, weight, ignore, and rowid columns at least eventually.
Need to support index syntax for columns, such as 0-4., 5,10-*